packages/contracts/contracts/generator/GenArt721GeneratorV0.sol

```diff
@@ -0,0 +1,436 @@
+     * supported by the DependencyRegistry or if the script storage version is
+     * not supported.
+     */
+    function getDependencyScriptBytes(
+        bytes32 dependencyNameAndVersion
+    ) internal view returns (bytes memory) {
+        uint256 scriptCount = dependencyRegistry.getDependencyScriptCount(
+            dependencyNameAndVersion
+        );
+
+        if (scriptCount == 0) {
+            return "";
+        }
+
+        address[] memory scriptBytecodeAddresses = new address[](scriptCount);
+
+        for (uint256 i = 0; i < scriptCount; i++) {
+            scriptBytecodeAddresses[i] = dependencyRegistry
+                .getDependencyScriptBytecodeAddress(
+                    dependencyNameAndVersion,
+                    i
+                );
+        }
+
+        bytes32 storageVersion = BytecodeStorageReader
+            .getLibraryVersionForBytecode(scriptBytecodeAddresses[0]);
+
+        uint256 offset;
+        if (storageVersion == BytecodeStorageReader.V0_VERSION_STRING) {
+            offset = 104;
+        } else if (storageVersion == BytecodeStorageReader.V1_VERSION_STRING) {
+            offset = 65;
+        } else {
+            revert("Unsupported storage version");
+        }
+
+        // @dev We concatenate the script chunks together with assembly
+        // in AddressChunks for gas efficiency.
+        return AddressChunks.mergeChunks(scriptBytecodeAddresses, offset);
+    }
+
+    /**
+     * @notice Get the dependency script for a given dependency name and version string.
+     * @param dependencyNameAndVersion The name and version of the dependency
+     * to retrieve the script for.
+     * @return The dependency script stiched together from all of the script chunks as a string.
+     * @dev This function will revert if the dependency name and version is not
+     * supported by the DependencyRegistry or if the script storage version is
+     * not supported.
+     */
+    function getDependencyScript(
+        string memory dependencyNameAndVersion
+    ) external view returns (string memory) {
+        bytes memory dependencyScript = getDependencyScriptBytes(
+            dependencyNameAndVersion.stringToBytes32()
+        );
+
+        return string(dependencyScript);
+    }
+
+    /**
+     * @notice Get the project script for a given core contract address and
+     * project ID.
+     * @param coreContractAddress The address of the core contract to retrieve
+     * the project script from.
+     * @param projectId The ID of the project to retrieve the script for.
+     * @return The project script stiched together from all of the script chunks.
+     * @dev This function will revert if the core contract address is not
+     * supported by the DependencyRegistry or if the contract at the address
+     * does not implement the IGenArt721CoreProjectScriptV0 or IGenArt721CoreProjectScriptV1.
+     */
+    function getProjectScriptBytes(
+        address coreContractAddress,
```

Comment:
super minor nit: could name this `coreContract`, as address is implied by the type (applies to the whole contract). This could also be done in a future upgrade to expedite shipping this (or ignored enirely!)